### PR TITLE
Improve background job error message

### PR DESCRIPTION
**Pull request by MorrisJobke**
@karlitschek I would like to backport this because it helps to make things like #19807 clear on the very first look and you don't have to guess which job and what could be possible wrong with it.

### CODE REVIEW

**Comment by MorrisJobke** [comment]
cc @rullzer @Xenopathic @LukasReschke @icewind1991

File: lib/private/backgroundjob/job.php [hunk 1]
```diff
@@ -52,7 +52,7 @@ public function execute($jobList, ILogger $logger = null) {
 			$this->run($this->argument);
 		} catch (\Exception $e) {
 			if ($logger) {
-				$logger->error('Error while running background job: ' . $e->getMessage());
+				$logger->error('Error while running background job (class: ' . get_class($this) . ', arguments: ' . print_r($this->argument, true) . '): ' . $e->getMessage());
```
Comment (DeepDiver1975) [inline lib/private/backgroundjob/job.php]: why not use logexception?

Comment (MorrisJobke) [inline lib/private/backgroundjob/job.php]: Good question - but I would also like to pass in more context. I will give it a try.

Comment (MorrisJobke) [inline lib/private/backgroundjob/job.php]: Updated - @DeepDiver1975 Please check out

File: lib/private/backgroundjob/job.php [hunk 2]
```diff
@@ -52,7 +52,8 @@ public function execute($jobList, ILogger $logger = null) {
 			$this->run($this->argument);
 		} catch (\Exception $e) {
 			if ($logger) {
-				$logger->error('Error while running background job: ' . $e->getMessage());
+				$logger->error('Error while running background job (class: ' . get_class($this) . ', arguments: ' . print_r($this->argument, true) . ')');
+				$logger->logException($e);
```
Comment (MorrisJobke) [inline lib/private/backgroundjob/job.php]: logException() will log the method like this: `ClassName->run(array)` which doesn't give any hint at the actual arguments. Therefore I log the arguments in a separate log entry.

Comment (DeepDiver1975) [inline lib/private/backgroundjob/job.php]: we might want to extend logException then to have a special key in the context array which can hold additional information being logged?

Comment (DeepDiver1975) [inline lib/private/backgroundjob/job.php]:
```php
$msg = 'Error while running background job (class: ' . get_class($this) . ', arguments: ' . print_r($this->argument, true) . ')';
$logger-logException($e, ['message' => $msg, 'app' => background])

**Comment by karlitschek** [comment]
backport is fine once everyone is happy here :+1:

**Comment by DeepDiver1975** [comment]
@MorrisJobke mind taking a look?

**Comment by PVince81** [comment]
Tested, works :+1: (manually throw an exception in ExpireSharesJob, run cron then check the log)

**Pull request merged**